mesonbuild/interpreter/interpreter.py

```diff
@@ -2242,6 +2242,14 @@ def make_test(self, node: mparser.BaseNode,
         if kwargs['timeout'] <= 0:
             FeatureNew.single_use('test() timeout <= 0', '0.57.0', self.subproject, location=node)
 
+        if kwargs['should_fail'] is not None and kwargs['expected_fail'] is not None:
+            raise InvalidArguments('Tried to use both \'should_fail\' and \'expected_fail\'')
+        elif kwargs['should_fail'] is not None:
+            mlog.deprecation('Use expected_fail instead of should_fail.', location=node)
```

Comment:
Actually hold on, this is already covered in the KwargInfo handling via
```python
deprecated='1.6.0', deprecated_message='Use expected_fail instead of should_fail'),
```

Reply:
I removed it, I thought KwargInfo handling didn't give exact location at first